gen/suites/vm_violations/address_resolution.go

```diff
@@ -0,0 +1,55 @@
+package main
+
+import (
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/specs-actors/actors/abi"
+	"github.com/filecoin-project/specs-actors/actors/builtin"
+	"github.com/filecoin-project/specs-actors/actors/abi/big"
+	"github.com/filecoin-project/specs-actors/actors/runtime/exitcode"
+
+	"github.com/filecoin-project/test-vectors/chaos"
+	. "github.com/filecoin-project/test-vectors/gen/builders"
+)
+
+func actorResolutionIdentity(v *Builder) {
+	v.Messages.SetDefaults(GasLimit(1_000_000_000), GasPremium(1), GasFeeCap(200))
+
+	alice := v.Actors.Account(address.SECP256K1, abi.NewTokenAmount(1_000_000_000_000))
+	v.CommitPreconditions()
+
+	v.Messages.Raw(alice.ID, chaos.Address, chaos.MethodResolveAddress, MustSerialize(&builtin.SystemActorAddr), Nonce(0), Value(big.Zero()))
+
+	invalidIDAddr, _ := address.NewIDAddress(77)
+	v.Messages.Raw(alice.ID, chaos.Address, chaos.MethodResolveAddress, MustSerialize(&invalidIDAddr), Nonce(1), Value(big.Zero()))
+	v.CommitApplies()
+
+	v.Assert.EveryMessageResultSatisfies(ExitCode(exitcode.Ok))
```

Comment:
Check that the response contains the input identity address?

Reply:
The predicate only works on test vectors that make a single message. for these, we expect different outputs for the 2 messages. worth splitting into different tests?

Reply:
(i'm less worried about the asserts doing 'full validation', because the actual validation when later running test vectors will just be exact equivalency of receipts as i understand it, while these asserts are more there to ensure is sane, and are lotus specific. We believe the spec-actors implementation used by lotus to already be tested, so spending a bunch of time on predicates / assertions to test that at generation time is less useful than creating the vectors with receipts that can be used by other actors implementations to test their implementations. maybe that's not the right way to be thinking about this, though?)

Reply:
So these assertions are really sanity checks. I kinda understand that because we've implemented the Chaos Actor behaviour that we're triggering now, we have a good sense that it's working as intended, but it's still worth doing these asserts for pragmatism.

The receipt comparison helps us pinpoint changes, but if the second message happens to be returning `t88` instead of `t77` (for whatever reason we didn't anticipate), we would've never caught it.

I really think we should add those assertions here.